mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/NodeMappingFactory.java

```diff
@@ -208,14 +210,36 @@ private static XContentBuilder getMappingContent() {
                                     .endObject()
                                 .endObject()
                             .endObject()
-                            
+
+                            .startObject(JsonMessageConstants.TEXT_BODY)
+                                .field(TYPE, STRING)
+                                .startObject(FIELDS)
+                                    .startObject(RAW)
+                                        .field(TYPE, STRING)
+                                        .field(ANALYZER, IndexCreationFactory.CASE_INSENSITIVE)
+                                        .field(IGNORE_ABOVE, LUCENE_LIMIT)
+                                    .endObject()
+                                .endObject()
+                            .endObject()
+
+                            .startObject(JsonMessageConstants.HTML_BODY)
+                                .field(TYPE, STRING)
+                                .startObject(FIELDS)
+                                    .startObject(RAW)
+                                        .field(TYPE, STRING)
+                                        .field(ANALYZER, IndexCreationFactory.CASE_INSENSITIVE)
+                                        .field(IGNORE_ABOVE, LUCENE_LIMIT)
+                                    .endObject()
+                                .endObject()
+                            .endObject()
 
                             .startObject(JsonMessageConstants.TEXT)
                                 .field(TYPE, STRING)
                                 .startObject(FIELDS)
                                     .startObject(RAW)
                                         .field(TYPE, STRING)
                                         .field(ANALYZER, IndexCreationFactory.CASE_INSENSITIVE)
+                                        .field(IGNORE_ABOVE, LUCENE_LIMIT)
```

Comment:
While coding unit-test, i have encountered this issue; which is a Lucene limitation:
http://stackoverflow.com/questions/24019868/utf8-encoding-is-longer-than-the-max-length-32766